src/plugins/intel_gpu/src/plugin/ops/reshape.cpp

```diff
@@ -24,7 +24,9 @@ static void CreateCommonReshapeOp(ProgramBuilder& p, const std::shared_ptr<ov::N
     auto input_pshape = op->get_input_partial_shape(0);
     auto output_pshape = op->get_output_partial_shape(0);
 
-    if (p.use_new_shape_infer() || op->is_dynamic()) {
+    if (p.use_new_shape_infer()
+        || op->is_dynamic()
+        || (p.get_engine().get_device_info().supports_immad && input_pshape.size() != output_pshape.size())) {
```

Comment:
Why don't you add any functional tests for the issued case? thanks.

Reply:
Is this okay with rank < 4 ?